process/smartContract/process.go

```diff
@@ -566,59 +604,63 @@ func (sc *scProcessor) processSCOutputAccounts(outputAccounts []*vmcommon.Output
 			//SC with data variables
 			err := sc.accounts.SaveDataTrie(acc)
 			if err != nil {
-				return nil, err
+				return err
 			}
 		}
 
+		// change code if there is a change
 		if len(outAcc.Code) > 0 {
 			err = sc.accounts.PutCode(acc, outAcc.Code)
 			if err != nil {
-				return nil, err
+				return err
 			}
 
-			//TODO remove this when receipts are implemented
-			log.Info(fmt.Sprintf("*** Generated/called SC account: %s ***", hex.EncodeToString(outAcc.Address)))
+			fmt.Printf("Created SC addess %s \n", hex.EncodeToString(outAcc.Address))
 		}
 
-		if outAcc.Nonce == nil || outAcc.Nonce.Cmp(big.NewInt(int64(acc.GetNonce()))) < 0 {
-			return nil, process.ErrWrongNonceInVMOutput
+		// change nonce only if there is a change
+		if outAcc.Nonce > 0 {
```

Comment:
might change this to `outAcc.Nonce != acc.GetNonce()` as to execute the instructions only when there is a change. Otherwise the comment is misleading